src/map/modes/political.hpp

```diff
@@ -10,9 +10,27 @@ std::vector<uint32_t> political_map_from(sys::state& state) {
 		auto fat_id = dcon::fatten(state.world, prov_id);
 		auto id = fat_id.get_nation_from_province_ownership();
 		uint32_t color = 0;
-		if(bool(id))
+		if(bool(id)) {
 			color = id.get_color();
-		else // If no owner use default color
+			switch(state.user_settings.vassal_color) {
+			case sys::map_vassal_color_mode::inherit:
+				//colour is interpolated from the owner and saved
+				break;
+			case sys::map_vassal_color_mode::same:
+				auto ovr = state.world.nation_get_overlord_as_subject(id);
+				if(ovr) {
+					auto ruler = state.world.overlord_get_ruler(ovr);
+					color = state.world.nation_get_color(id);
+				}
+				break;
+			case sys::map_vassal_color_mode::none:
+				auto ovr = state.world.nation_get_overlord_as_subject(id);
+				if(ovr) {
+					color = state.world.national_identity_get_color(state.world.nation_get_identity_from_identity_holder(id));
```

Comment:
or rather, this won't work, because determining the color is more complicated than that (it depends partly on the nation's current government type). Furthermore, a common asked-for extension is for effects that can change a nation's color. Which, this would prevent from working.

Reply:
I suggest moving the vassal coloring logic out of the update political function and move it here, possibly with caching if it is too expensive. Then it could be turned on and off trivially